### DISCUSSION TITLE
Custom Macro placement detailed route issue: Maze Route cannot find path of net

### DISCUSSION THREAD

**Discussion by khaledhasanzami** [The-OpenROAD-Project/OpenROAD#3044] [General]
I am using a custom macro placement script. Changing the macro position from default to any other formation causing error in detailed routing. 

`[ERROR DRT-0255] Maze Route cannot find path of net (some net) in worker of routeBox (34000 75000) (41000 45000)`

Is it an algorithm error, any idea?

**Comment by vijayank88** [comment]
Is automatic macro placement creating that error?

**Reply by khaledhasanzami** [reply]
No, I had custom macro placement

**Reply by vijayank88** [reply]
Then you need place macros on right position.
Try once with automatic macro placement and check its coordinates. If automatic macro placement causes any DRT issues, share the test case.
As Matt asked test case, you can provide your custom macro placement test to check the issue.

**Reply by vijayank88** [reply]
@khaledhasanzami Did you tried?

**Comment by maliberty** [comment]
This is a problem in detailed routing that would require a test case to investigate.  Can you provide one?

**Comment by maliberty** [comment]
I don't expect this error from the router even with a bad macro placement.

**Comment by msingh9** [comment]
How do you do custom macro placement?

**Comment by maliberty** [comment]
It would be great to have a test case for this problem as is so we can fix it in addition to any workarounds in placement to avoid it.
